src/include/wiredtiger.in

```diff
@@ -2109,6 +2131,48 @@ struct __wt_connection {
 	/*! @} */
 
 	/*!
+	 * @name Transactions
+	 * @{
+	 */
+	/*!
+	 * Query the global transaction timestamp state.
+	 *
+	 * @snippet ex_all.c query timestamp
+	 *
+	 * @param connection the connection handle
+	 * @param[out] timestamp a buffer that will be set to the timestamp
+	 * being queried.  Must be large enough to hold a hex-encoded timestamp
+	 * (i.e., double the timestamp size plus one byte for NUL termination).
+	 * @configstart{WT_CONNECTION.query_timestamp, see dist/api_data.py}
+	 * @config{get, specify which timestamp to query: \c all_committed
+	 * returns the largest timestamp such that all earlier timestamps have
+	 * committed.  See @ref transaction_timestamps., a string\, chosen from
+	 * the following options: \c "all_committed"; default \c all_committed.}
+	 * @configend
+	 * @errors
+	 */
+	int __F(query_timestamp)(
+	    WT_CONNECTION *connection, char *timestamp, const char *config);
```

Comment:
I've created a type for internal, raw timestamps (which were consistently `uint8_t` arrays before.  The `char *` buffers are hexadecimal timestamp strings.  I've changed parameter names to clarify.